VulkanHppGenerator.cpp

```diff
@@ -3144,7 +3163,7 @@ void VulkanHppGenerator::appendHandle( std::string & str, std::pair<std::string,
     assert( commandIt != m_commands.end() );
     for ( auto const & parameter : commandIt->second.params )
     {
-      if ( handleData.first != parameter.type.type )  // the commands use this handleData type !
+      if ( isHandleType( parameter.type.type ) && ( handleData.first != parameter.type.type ) )  // the commands use this handleData type !
```

Comment:
There's an implicit behavior change here that isn't explained by the commit title.  Was it intentional?  Can you elaborate on it?

Reply:
Also, completely obliquely, I noticed from this change that **Vulkan-Hpp** seems to generate another header file `vulkan_raii.hpp` that is *not* currently included in **KhronosGroup/Vulkan-Headers** and is not shipped in the **Vulkan SDK**.  Should it be?

Reply:
That code change is needed for the separation into multiple files.
Before this change, structure and handle wrapper classes were listed interleaved in vulkan.hpp. With this change, they are separated into two files, vulkan_structs.hpp and vulkan_handles.hpp.
In this part of the code, I determine which handles need to be listed first (any handle A, that is part of any command of handle B needs to be listed before handle B). But there are also non-handles (like structs) part of the commands, that should not be listed here. Therefore I check for handles here.

Reply:
vulkan_raii.hpp introduces additional functionalities (a set of raii-compliant handle wrapper classes) and is "on-top-of" vulkan.hpp.
It is not (yet) part of Vulkan-Headers, but I plan to add it soon.
If it's substiantially easier to add it as well right now, I think that would be fine.